airgun/views/contentviewfilter.py

```diff
@@ -0,0 +1,343 @@
+    """
+    edit = Text(".//button[contains(@ng-click, 'rule.editMode')]")
+    save = Text(".//button[contains(@ng-click, 'handleSave()')]")
+    cancel = Text(".//button[contains(@ng-click, 'handleCancel()')]")
+
+
+class CVFRuleVersion(View):
+    """'Version' column for content view filter rule. Depending on type (e.g.
+    'Equal To', 'Greater Than' etc) can have different set of inputs.
+    """
+    rule_type = SatSelect(locator=".//select[@ng-model='rule.type']")
+    rule = ConditionalSwitchableView(reference='rule_type')
+    version_text = Text(
+        ".//span[@ng-hide='rule.editMode']//descendant::span[last()]")
+
+    @rule.register('All Versions')
+    class all_versions(View):
+        pass
+
+    @rule.register('Equal To')
+    class equal_to(View):
+        version = TextInput(id='version')
+
+    @rule.register('Greater Than')
+    class greater_than(View):
+        min_version = TextInput(id='minVersion')
+
+    @rule.register('Less Than')
+    class less_than(View):
+        max_version = TextInput(id='maxVersion')
+
+    @rule.register('Range')
+    class range(View):
+        min_version = TextInput(id='minVersion')
+        max_version = TextInput(id='maxVersion')
+
+    def fill(self, values):
+        """Custom fill to support passing values for all inputs in single tuple
+        without the need to specify specific input name.
+
+        :param tuple values: tuple containing values for specific version type
+            and its inputs, e.g. `('Equal To', '0.5')` or
+            `('Range', '4.1', '4.6')`
+        """
+        if not isinstance(values, tuple):
+            values = (values,)
+        was_change = self.rule_type.fill(values[0])
+        if len(values) == 1:
+            return was_change
+        widgets_list = [
+            'rule.{}'.format(name) for name in self.rule.widget_names]
+        values = dict(zip(widgets_list, values[1:]))
+        return super().fill(values) or was_change
+
+    def read(self):
+        """Custom `read` to return "summary" text value, not the dict with
+        every included widget separately.
+        """
+        return self.version_text.read()
+
+
+class CVFEditableEntry(EditableEntry):
+    """Content view filter variant of Editable Entry, main difference of which
+    is ``span`` tags instead of ``dd`` and ``dt``.
+    """
+
+    def __init__(self, parent, locator=None, name=None, logger=None):
+        """Supports initialization via ``locator=`` or ``name=``"""
+        if locator and name or not locator and not name:
+            raise TypeError('Please specify either locator or name')
+        locator = (
+            locator or
+            ".//span[contains(@class, 'info-label')][normalize-space(.)='{}']"
+            "/following-sibling::span[contains(@class, 'info-value')]"
+            .format(name)
+        )
+        super(EditableEntry, self).__init__(parent, locator, logger)
+
+
+class AffectedRepositoriesTab(SatSecondaryTab):
```

Comment:
Sounds like bad idea to me to accept tab name and not to use it afterwards. But i can move `TAB_NAME` here since this class is specific for that one tab.

Reply:
no, I am about:
```
    TAB_LOCATOR = ParametrizedLocator(
        './/nav[@class="ng-scope" or not(@*)]/ul[contains(@class, "nav-tabs")]'
        '/li[./a[starts-with(normalize-space(), {@tab_name|quote})]]'
    )
```
Why do you need that complex locator, if you know exactly name for your tab